### PR TITLE
Providing haptic feedback when the preview view is ready to send.

### PR DESCRIPTION
**Pull request by MartinJibber**
This does not include work to make the text field white when active.

https://user-images.githubusercontent.com/88675954/151258260-8c4ddc0a-85b0-45bf-959e-7369f6384937.MP4

### CODE REVIEW

File: Shared/UI/Transitions/Transtion Types/TransitionRouter+Message.swift [hunk 1]
```diff
@@ -33,15 +33,15 @@ extension TransitionRouter {
 
         if let message = fromView.message {
             snapshot.configure(with: message)
-            snapshot.bubbleView.bubbleColor = fromView.bubbleView.bubbleColor
+            snapshot.bubbleView.setBubbleColor(fromView.bubbleView.bubbleColor, animated: false)
```
Comment (MartinJibber) [inline Shared/UI/Transitions/Transtion Types/TransitionRouter+Message.swift:36]: You can know animate the bubble color.

File: Shared/UI/Views/SpeechBubbleView.swift [hunk 2]
```diff
@@ -33,13 +33,6 @@ class SpeechBubbleView: BaseView {
             guard let cgColor = self.bubbleLayer.fillColor else { return nil }
             return UIColor(cgColor: cgColor)
         }
-        set {
-            // CALayers are implicitly animated. Disable animations so the color changes immediately.
-            CATransaction.begin()
-            CATransaction.setDisableActions(true)
-            self.bubbleLayer.fillColor = newValue?.cgColor
-            CATransaction.commit()
-        }
```
Comment (MartinJibber) [inline Shared/UI/Views/SpeechBubbleView.swift:42]: Previously this was the only way to set the bubble color and it prevented animations. I made a set function down below.

File: Shared/UI/Views/SpeechBubbleView.swift [hunk 3]
```diff
@@ -119,6 +112,15 @@ class SpeechBubbleView: BaseView {
...
 
+    func setBubbleColor(_ color: UIColor?, animated: Bool) {
+        // CALayers are implicitly animated.
+        // If specified, disable animations so the color changes immediately.
+        CATransaction.begin()
+        CATransaction.setDisableActions(!animated)
+        self.bubbleLayer.fillColor = color?.cgColor
+        CATransaction.commit()
+    }
+
```
Comment (MartinJibber) [inline Shared/UI/Views/SpeechBubbleView.swift:123]: here

File: iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift [hunk 4]
```diff
@@ -394,6 +397,21 @@ class SwipeableInputAccessoryView: BaseView, UIGestureRecognizerDelegate, Active
...
+                previewView.setBubbleColor(ThemeColor.D1.color, animated: true)
+                self.impactFeedback.impactOccurred()
+            }
+            self.isPreviewInDropZone = true
+        } else {
+            if self.isPreviewInDropZone {
+                previewView.setBubbleColor(ThemeColor.B1.color, animated: true)
+            }
+            self.isPreviewInDropZone = false
+        }
```
Comment (MartinJibber) [inline iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift:414]: Change the color and provide haptic feedback when the bubble moves into the drop zone.

**Review by Benjamindodgson — APPROVED** [review]

**Pull request merged**